servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1750,6 +1777,63 @@ private void setStateFromEntry(SetStateEntry entry) throws FileDoesNotExistExcep
   }
 
   /**
+   * Checks user's permission on a path
+   *
+   * @param action requested {@link FileSystemAction} by user
+   * @param path the path to check permission on
+   * @param checkParent indicates whether to check the path or its parent
+   * @throws AccessControlException if permission checking fails
+   * @throws InvalidPathException if the path is invalid
+   */
+  private void checkPermission(FileSystemAction action, TachyonURI path,
+      boolean checkParent) throws AccessControlException, InvalidPathException {
+    // bypasses permission checking if security is not enabled.
+    if (MasterContext.getConf().get(Constants.SECURITY_AUTHENTICATION_TYPE).equals("NOSASL")
```

Comment:
Shall we add a util method checking if security is enabled, similarly to how we check if lineage is enabled?